core/core.go

```diff
@@ -557,6 +630,103 @@ func (t *Textile) noAuthFunc(ctx context.Context) (context.Context, error) {
 	return ctx, nil
 }
 
+func generatePowUnaryInterceptor(serviceName string, allowedMethods []string, serviceDesc *desc.ServiceDescriptor, stub *grpcdynamic.Stub, pc *powc.Client, c *mdb.Collections) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		methodParts := strings.Split(info.FullMethod, "/")
+		if len(methodParts) != 3 {
+			return nil, status.Errorf(codes.Internal, "error parsing method string %s", info.FullMethod)
+		}
+		methodServiceName := methodParts[1]
+		methodName := methodParts[2]
+
+		if methodServiceName != serviceName {
+			return handler(ctx, req)
+		}
+
+		if pc == nil {
+			return nil, status.Error(codes.Internal, "Powergate isn't enabled in Hub")
+		}
+
+		isAllowedMethod := false
+		for _, method := range allowedMethods {
+			if method == methodName {
+				isAllowedMethod = true
+				break
+			}
+		}
```

Comment:
Now we know we're dealing with a request for the service we're configured to be handling, make sure the request method is allowed.